### ISSUE TITLE
the calculated position is wrong when there is an margin applied to main content area. Is there a manual offset override? 

### ISSUE THREAD

**Issue by xinghengwang** [gilbarbara/react-joyride#164]
Hi, 

If the position is off by 60 or 80 px (of the tool tip).  What could be the cause? 

Thanks
=Xing

**Comment by IanVS** [comment]
Do you have fixed-position elements on your page, at a higher level in the DOM than where your `Joyride` is mounted?

**Comment by xinghengwang** [comment]
indeed. So I can't really avoid that because I have a navibar on top, if I don't offset the main content area down a little, the top of the main content area will be covered by the navibar.
Is there anyway in Joyride to add an offset to compensate?

**Comment by IanVS** [comment]
What I did was to mount the joyride at the top of my body.

pseudo-code:

```html
<body>
    <header style='position: fixed;'>navigation</header>
    <div id='YourAppHere'></div>
    <Joyride ref={c => joyride = c} />
</body>
```

That will put a `joyride` variable on the global scope, which you can then access in the rest of your app.  There are better ways to do that, but hopefully you get the idea.  It essentially puts the joyride into a portal, outside of the rest of your app.  Let me know if that doesn't make sense.

**Comment by xinghengwang** [comment]
Thanks for the tip. I'll try it out.

**Comment by xinghengwang** [comment]
hm... 
still doesn't work. 
so this is what I did: 

My App render code is like this: 

```
render() {
   const styles = require('./App.scss');

    return (
      <div>
        <HeaderBar/>

        <div className={styles.appContent}>
          {this.props.children}
        </div>

        <footer className="container-fluid text-center">
        </footer>
        <JoyrideWithRedux/>
      </div>
    );
}
```

The scss for App.scss have this: 

```
.appContent {
  margin: 64px 0; // for fixed navbar
}
```

Of course, I now know this style margin offset is what is causing the issue. since if I remove this style, the Joyride's positions lines up correctly. 

However, when I remove this, the Navigation bar will cover up the main content area for appContent. So I really want to keep it in there.  

For Joyride, I already mounted outside of the appContent section. 

(Btw, the JoyrideWithRedux is just a simple wrapper of Joyrider for redux connect to the store where I can store all the steps without pass the joyride object as a global variable. 

Any help will be appreciated.
